ggdeploymentd/src/deployment_handler.c

```diff
@@ -1977,9 +1990,188 @@ static GglError wait_for_deployment_status(GglMap resolved_components) {
     return GGL_ERR_OK;
 }
 
-// This will be refactored soon with recipe2unit in c, so ignore this warning
-// for now
-// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+static GglError apply_reset_config(
+    GglBuffer component_name, GglMap component_config_map
+) {
+    GglObject *reset_configuration = NULL;
+    GglError ret = ggl_map_validate(
+        component_config_map,
+        GGL_MAP_SCHEMA(
+            { GGL_STR("reset"), false, GGL_TYPE_LIST, &reset_configuration }
+        )
+    );
+    if (ret != GGL_ERR_OK) {
+        return ret;
+    }
+
+    // If there is no reset configuration, then there is no
+    // configuration update to make
+    if (reset_configuration != NULL) {
```

Comment:
nit: Let's check for `reset_configuration == NULL` and then return if it does. Saves us from adding indentations to the code